dlt/destinations/dataset/relation.py

```diff
@@ -213,72 +243,150 @@ def __init__(
         limit: int = None,
         selected_columns: Sequence[str] = None,
```

Comment:
we don't need selected columns and limit anymore, you can store this info in the query object.